package/libs/openssl/Makefile

```diff
@@ -184,7 +190,7 @@ define Build/Configure
 			--openssldir=/etc/ssl \
 			$(TARGET_CPPFLAGS) \
 			$(TARGET_LDFLAGS) -ldl \
-			-DOPENSSL_SMALL_FOOTPRINT \
+			$(if $(CONFIG_OPENSSL_OPTIMIZE_SPEED),,-DOPEN_SSL_SMALL_FOOTPRINT) \
```

Comment:
A typo `OPEN_SSL_` to `OPENSSL_`.  We can also move this to `TARGET_CPPFLAGS` in the same spirit of `-Os` for `TARGET_CFLAGS`.

BTW, It will be interesting to know two things with the change
 - What's the stats when only `-O3` was enabled, or only `SMALL_FOOTPRINT` was disabled
 - What's the impact on size with the change

Reply:
@yousong 
Thanks for having a look! Fixed the typo and as for size...

Default optimization (binary / ipk):
1.37M / 629 kbyte

Optimized for speed (binary / ipk):
1.72M / 810 kbyte

Unfortunately I don't have time to test more variants for now.